### ISSUE TITLE
General Comments

### ISSUE THREAD

**Issue by ibirnam** [cs10/bjc-r#195]
I just emailed these to Peter, but comments 1 and 4 were based off of a previous version of the lab. Though why did we scrap the variables section? They need to know it before HW1, and this lab is their only chance to learn it.
1. [ ] We should probably define (or remind them) what a script is, as its used a lot in the first page. This can be something that is added to the lab, or gone over by the TA beforehand, but just wanted to bring it up.
2. [ ] on the If and If Else page, it would good to clarify what you mean by "temporarily". For example, you could say that a value that is stored in the "answer" variable is kept there only until the next time the ask block is called.
3. [x]  LOVE the page on and, or, not, especially the truth tables, I'm glad they're learning this early on.
4. [ ] This is more a suggestion, but perhaps add quiz question that has both global and script variables involved? A basic scope question would be pretty beneficial
5. [ ] On the Practice With Complex Booleans page the 2nd bullet point I think could be reworded for clarity to say "You could write this without using your <weekend?()> block, but it would be much easier if you use it"
6. [ ] On the same page, bullet 3 asks them to write a block we just showed them how to do on the previous page. Seems a bit redundant.
7. [x] This link for predicate solutions leads to a 405 page: http://peterasujan.github.io/bjc-r/cur/programming/functions/predicates/ge-soln.html

**Comment by xtitter** [comment]
Good comments.  A few thoughts:

2) we shouldn't call (answer) a variable, but a reporter, since it can't be set. Brian Harvey is ... adamant about this.
4) no global variables please!  Not until lab 4.

**Comment by ibirnam** [comment]
Is there a reason though that the page on variables was taken out? They need to be taught what a script variable is, at least.

**Comment by ibirnam** [comment]
also i realized my link for number 7 didnt go through, i updated it accordingly

**Comment by peterasujan** [comment]
> Is there a reason though that the page on variables was taken out? They need to be taught what a script variable is, at least.

Yeah, we're planning on teaching them when they're needed (a bit later in the lab).

**Comment by xtitter** [comment]
Peter is going to add script variables for sure, and set.  Supposed to be in lab 3.  But, it seemed tacked on at the front; we wanted it integrated for students when they needed it, which is in custom-block predicates I think.

**Comment by ibirnam** [comment]
Ah ok got it, I was just confused because it jumped from the "More Complex Boolean Expressions" page right into the quiz abut script variables, but if the page explaining variables is gonna be added in, then that works out

**Comment by cycomachead** [comment]
Anything left to be resolved here?
